### DISCUSSION TITLE
Updating dynamic variable in loop

### DISCUSSION THREAD

**Discussion by imakeitworknet** [statamic/cms#7285] [Q&A] [answered]
Hi. I have a page with Related Entries (bard block Entries) in it. Like:

```
related_entries:
          - 69677794-6d85-496f-b22c-ab054c2f269e
          - 43e250f2-df2c-452f-828b-54e4383dcd41
```
          
I also have a dynamic variable named {{ found = 0 }} in the antlers template. Is there a way to update that variable while looping through related entries, and getting it later in the template? Here's complete example:

```
<section>
  {{ found = 0 }}
       {{ related_entries }}
           {{ if SOME LOGIC }}
                  {{ found = 1 }}
           {{ /if }}
       {{ /related_entries }}

{{ if found==1 }}
   /* do something */
{{ /if }}
</section>
```

Currently everything what happens in {{ related_entries }} loop stays in it, so {{ found }} variable does not get through.
Thanks!

**Comment by jackmcdade** [comment]
Are you using the new (runtime) parser?

**Reply by imakeitworknet** [reply]
Yes. And variable works and updates outside of {{ related_entries }} loop. But if I update it's value in that {{ related_entries }} loop, after the loop it still shows old value.

**Reply by jackmcdade** [reply]
Okay gotcha – seems a bit like a bug, gonna open it up and we'll take a look! https://github.com/statamic/cms/issues/7297

**Comment by jackmcdade** [comment, marked as answer]
Turns out it was just a classic PEBKAU. All good.
